pkg/ccl/backupccl/restore_span_covering_test.go

```diff
@@ -28,21 +33,41 @@ import (
 
 // MockBackupChain returns a chain of mock backup manifests that have spans and
 // file spans suitable for checking coverage computations. Every 3rd inc backup
-// introduces a span and drops a span. Incremental backups have half as many
-// files as the base. Files spans are ordered by start key but may overlap.
+// reintroduces a span. On a random backup, one random span is dropped and
+// another is added. Incremental backups have half as many files as the base.
+// Files spans are ordered by start key but may overlap.
 func MockBackupChain(length, spans, baseFiles int, r *rand.Rand) []backuppb.BackupManifest {
 	backups := make([]backuppb.BackupManifest, length)
 	ts := hlc.Timestamp{WallTime: time.Second.Nanoseconds()}
+	backupWithDroppedSpan := r.Intn(len(backups))
+	spanIdxToDrop := r.Intn(spans)
+
+	genTableID := func(j int) uint32 {
+		return uint32(10 + j*2)
+	}
+
 	for i := range backups {
 		backups[i].Spans = make(roachpb.Spans, spans)
+		backups[i].IntroducedSpans = make(roachpb.Spans, 0)
 		for j := range backups[i].Spans {
-			backups[i].Spans[j] = makeTableSpan(uint32(100 + j + (i / 3)))
+			tableID := genTableID(j)
+			backups[i].Spans[j] = makeTableSpan(tableID)
 		}
 		backups[i].EndTime = ts.Add(time.Minute.Nanoseconds()*int64(i), 0)
 		if i > 0 {
 			backups[i].StartTime = backups[i-1].EndTime
+
+			if i >= backupWithDroppedSpan {
+				// Drop a span present in the first i backups, and add a new one
```

Comment:
So this will drop a span in every incremental layer after `backupWithDroppedSpan`?